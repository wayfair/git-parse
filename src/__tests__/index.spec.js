import { gitToJs } from "../index";
import path from "path";
import { renameSync } from "fs";

describe("index", () => {
  const pathToTestRepo = path.join(__dirname, "git_repo");
  const gitRepoToCommit = path.join(pathToTestRepo, "dot_git");
  const realDotGitDir = path.join(pathToTestRepo, ".git");

  it("should reject the promise if the directory is not the root of a git repo", () => {
    return expect(() => gitToJs(pathToTestRepo)).rejects.toThrow(
      /does not appear to be the root of a git repository/
    );
  });

  it("should reject the promise if the directory does not exist", () => {
    return expect(() =>
      gitToJs("/some_crazy_directory_that_does_not_exist/asdfasdfasdfasdfxxxx")
    ).rejects.toThrow(/was not found/);
  });

  xit("VISUAL GREP TEST ONLY, ENABLE FOR DEBUGGING PURPOSES ONLY", (done) => {
    const repoPath = ".";

    gitToJs(repoPath).then((data) => {
      console.log(data);
      expect(true).toBe(true);
      done();
    });
  });

  describe("tests with a real repository", () => {
    beforeAll(() => {
      renameSync(gitRepoToCommit, realDotGitDir);
    });

    it("reads a git repo successfully", async () => {
      const parsedRepo = await gitToJs(pathToTestRepo);
      expect(parsedRepo).toMatchSnapshot();
    });

    afterAll(() => {
      renameSync(realDotGitDir, gitRepoToCommit);
    });
  });
});
