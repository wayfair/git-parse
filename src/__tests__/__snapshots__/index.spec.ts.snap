// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index tests with a real repository reads a git repo successfully 1`] = `
[
  {
    "authorEmail": "mpackard@wayfair.com",
    "authorName": "mpackard@wayfair.com",
    "date": "Thu, 1 Feb 2018 16:35:05 -0500",
    "filesAdded": [
      {
        "path": "file.txt",
      },
    ],
    "filesDeleted": [],
    "filesModified": [],
    "filesRenamed": [],
    "hash": "99402395036c4282de837cd1d2004ee7b11988bb",
    "message": "added file",
  },
]
`;
